typing/typecore.ml

```diff
@@ -5826,8 +5890,15 @@ and type_expect_
           (* The "body" component of the scope escape check. *)
           unify_exp new_env body (newvar (Jkind.Builtin.any ~why:Dummy_jkind)))
       in
+      let exp =
+        match mutable_flag, pat_exp_list with
+        | Immutable, _ -> Texp_let(rec_flag, pat_exp_list, body)
+        | Mutable, [vb] -> Texp_letmutable(vb, body)
+        | Mutable, _ -> fatal_error "Typecore.type_expect"
+          (* Unreachable: should be prevented by [value_bindings_mutability] *)
```

Comment:
(Just a note to update this comment if you do change the name of `value_bindings_mutability`)